web/packages/teleterm/src/services/tshd/cloneableClient.ts

```diff
@@ -254,9 +254,45 @@ export type TshdRpcError = Pick<
   isResolvableWithRelogin: boolean;
 };
 
-/** Checks if the given value is a `TshdRpcError`. */
-export function isTshdRpcError(error: unknown): error is TshdRpcError {
-  return error['name'] === 'TshdRpcError';
+/**
+ * Subset of gRPC status codes.
+ * It doesn't contain UNAUTHENTICATED and ABORTED values, which could be
+ * mistakenly used as PERMISSION_DENIED and CANCELLED.
+ * @see https://grpc.io/docs/guides/status-codes
+ */
+type RpcStatusCode =
+  | 'CANCELLED'
+  | 'UNKNOWN'
+  | 'INVALID_ARGUMENT'
+  | 'DEADLINE_EXCEEDED'
+  | 'NOT_FOUND'
+  | 'ALREADY_EXISTS'
+  | 'PERMISSION_DENIED'
+  | 'RESOURCE_EXHAUSTED'
+  | 'FAILED_PRECONDITION'
+  | 'OUT_OF_RANGE'
+  | 'UNIMPLEMENTED'
+  | 'INTERNAL'
+  | 'UNAVAILABLE'
+  | 'DATA_LOSS';
+
+/**
+ * Checks if the given value is a `TshdRpcError`.
```

Comment:
I'd add a line explaining that it's meant to be used to check errors received from a gRPC client that crosses context bridge.

If someone looks at this function and they're not aware of how gRPC clients and context bridge interact together, it might be tempting to use this function for other purposes, e.g. somewhere within tshd events backend implementation. But that would just lead someone to a dead end.

Reply:
Right. I rephrased the messaged a bit:
>It's meant to be used to check errors received from a gRPC client that was produced with `cloneClient()`.

It's because we clone the gRPC client in the main process too (so we can check for `isResolvableWithRelogin` if we need to).